buildpacks/ruby/src/layers/bundle_download_layer.rs

```diff
@@ -1,161 +1,163 @@
-
-        log_step_timed(format!("Running {}", fmt::command(short_name)), || {
-            cmd.named_output().map_err(|error| {
-                fun_run::map_which_problem(error, cmd.mut_cmd(), self.env.get("PATH").cloned())
-            })
-        })
-        .map_err(RubyBuildpackError::GemInstallBundlerCommandError)?;
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub(crate) struct MetadataV1 {
+    pub(crate) version: ResolvedBundlerVersion,
+}
 
-        LayerResultBuilder::new(self.metadata.clone())
-            .env(
-                LayerEnv::new()
-                    .chainable_insert(Scope::All, ModificationBehavior::Delimiter, "PATH", ":")
-                    .chainable_insert(
-                        Scope::All,
-                        ModificationBehavior::Prepend,
-                        "PATH", // Ensure this path comes before default bundler that ships with ruby, don't rely on the lifecycle
-                        bin_dir,
-                    )
-                    .chainable_insert(Scope::All, ModificationBehavior::Delimiter, "GEM_PATH", ":")
-                    .chainable_insert(
-                        Scope::All,
-                        ModificationBehavior::Prepend,
-                        "GEM_PATH", // Bundler is a gem too, allow it to be required
-                        gem_path,
-                    ),
-            )
-            .build()
-    }
+#[derive(Debug, thiserror::Error)]
+pub(crate) enum MetadataError {
+    // Update if migrating between a metadata version can error
+}
 
-    fn existing_layer_strategy(
-        &mut self,
-        _context: &BuildContext<Self::Buildpack>,
-        layer_data: &LayerData<Self::Metadata>,
-    ) -> Result<ExistingLayerStrategy, RubyBuildpackError> {
-        let old = &layer_data.content_metadata.metadata;
-        let now = self.metadata.clone();
-        match cache_state(old.clone(), now) {
-            State::NothingChanged(_version) => {
-                log_step("Using cached version");
-
-                Ok(ExistingLayerStrategy::Keep)
-            }
-            State::BundlerVersionChanged(_old, _now) => {
-                log_step(format!(
-                    "Clearing cache {}",
-                    fmt::details("bundler version changed")
-                ));
+fn download_bundler(
+    bullet: Print<SubBullet<Stdout>>,
+    env: &Env,
+    metadata: &Metadata,
+    path: &Path,
+) -> Result<(Print<SubBullet<Stdout>>, LayerEnv), RubyBuildpackError> {
+    let bin_dir = path.join("bin");
+    let gem_path = path;
+
+    let mut cmd = Command::new("gem");
+    cmd.args(["install", "bundler"]);
+    cmd.args(["--version", &metadata.version.to_string()]) // Specify exact version to install
+        .env_clear()
+        .envs(env);
 
-                Ok(ExistingLayerStrategy::Recreate)
-            }
-        }
-    }
-}
+    let short_name = fun_run::display(&mut cmd); // Format `gem install --version <version>` without other content for display
```

Comment:
interesting way to hide subsequent args but why not just show those extra args to the user?

Reply:
When presented with a failed build, the average user might get distracted by those extra arguments when they should be looking somewhere else. If the command fails, it prints out all the arguments, which is when a user might actually want to see them. Effectively: I see shelling out to this command as an implementation detail. I considered hiding it all together, but I think showing the short version peels back a little of the magic without being overwhelming.

Versus with `bundle install` that's a system-critical load-bearing source of support tickets and I want developers to be able to copy and paste the EXACT thing we run into their local machine.